File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int ans = 0, cnt = 0;
        string p;

        for(int i=0;i<s.size();i++){
            int l = i;
            int r = i;
            while(l>=0 && r<s.size() && s[l]==s[r]){
                if(r-l+1 > ans){
                    ans = r - l + 1;
                    p = s.substr(l, r-l+1);
                }
                r++;
                l--;
                cnt++;
            }

            l = i;
            r = i+1;
            while(l>=0 && r<s.size() && s[l]==s[r]){
                if(r-l+1 > ans){
                    ans = r - l + 1;
                    p = s.substr(l, r-l+1);
                }
                r++;
                l--;
                cnt++;
            }
        }

        return cnt;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r = 0, 1
        ans = 0
        while r < len(prices):
            if prices[l] < prices[r]:
                ans = max(ans, prices[r]-prices[l])
                r += 1
            else:
                l = r
                r += 1
        
        return ans